utilities/SetupForDevelopment.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+#==========================================================================
+#
+#   Copyright NumFOCUS
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#          https://www.apache.org/licenses/LICENSE-2.0.txt
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+#==========================================================================*/
+
+
+# Run this script to set up the git hooks for committing changes to PCT.
+# For more information, see:
+#   https://www.itk.org/Wiki/ITK/Git#Hooks
+#   https://www.itk.org/Wiki/Git/Hooks
+
+#!/usr/bin/env bash
+#=============================================================================
+# Copyright 2010-2012 Kitware, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+# Run this script to set up local Git hooks for this project.
+
+# Project configuration instructions:
+#
+# - Publish a "hooks" branch in the project repository such that
+#   clones will have "refs/remotes/origin/hooks".
+#
+# - Populate adjacent "config" file with:
+#    hooks.url    = Repository URL publishing "hooks" branch
+#    hooks.branch = Repository branch instead of "hooks"
+
+egrep-q() {
+	egrep "$@" >/dev/null 2>/dev/null
+}
+
+die() {
+	echo 1>&2 "$@" ; exit 1
+}
+
+# Make sure we are inside the repository.
+cd "${BASH_SOURCE%/*}" &&
+
+# Select a hooks branch.
+if url=$(git config --get hooks.url); then
+	# Fetch hooks from locally configured repository.
+	branch=$(git config hooks.branch || echo hooks)
+elif git for-each-ref refs/remotes/origin/hooks 2>/dev/null |
+     egrep-q 'refs/remotes/origin/hooks$'; then
+	# Use hooks cloned from origin.
+	url=.. && branch=remotes/origin/hooks
```

Comment:
Can we use the RTK repo URL directly? (to address the use case where users use their fork as origin with only the master branch)

Reply:
It might be better to instead add the link as a last option in those checks? This allow the user to configure it's own hook branch but fallback on the PCT hook branch if none exist.